pyscript.core/src/plugins/py-terminal.js

```diff
@@ -135,14 +140,18 @@ const pyTerminal = async () => {
     } else {
         // in the main case, just bootstrap XTerm without
         // allowing any input as that's not possible / awkward
-        hooks.main.onReady.add(function main({ interpreter, io }) {
+        hooks.main.onReady.add(function main({ interpreter, io, run }) {
             console.warn("py-terminal is read only on main thread");
             hooks.main.onReady.delete(main);
-            init({
+
+            // on main, it's easy to trash and clean the current terminal
+            globalThis.__py_terminal__ = init({
                 disableStdin: true,
                 cursorBlink: false,
                 cursorStyle: "underline",
             });
+            run("from js import __py_terminal__ as __terminal__");
+            delete globalThis.__py_terminal__;
```

Comment:
You still ought to be able to assign to the Python scope directly from JavaScript rather than doing this `globalThis` dance...

Reply:
agreed, I am not proud of that ugly dance but it worked well in both MicroPython and Pyodide ... can I do better for both "worlds"? 🤔